File: frontend/types/report.ts
// academix_tz.md §2.2 "Hisobotlar".
export type ReportType = "SCHOOL" | "CLASS" | "STUDENT";

export interface Report {
  id: string;
  type: ReportType;
  quarter: string;
  targetId: string | null;
  generatedBy: string;
  generatedAt: string;
}

export interface GenerateReportRequest {
  type: ReportType;
  quarter: string;
  targetId?: string;
}

File: backend/src/main/java/uz/academixai/reporting/domain/Report.java
package uz.academixai.reporting.domain;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * academix_tz.md §2.2 {@code POST/GET /admin/reports} — no DDL exists in the spec (see V36
 * migration's comment); plain domain object, no framework annotations.
 */
public record Report(
    UUID id,
    UUID schoolId,
    ReportType type,
    String quarter,
    UUID targetId,
    String fileUrl,
    UUID generatedBy,
    LocalDateTime generatedAt) {}
